### PR TITLE
fix: mention suggestion list does not follow cursor position (AR-2936)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2936" title="AR-2936" target="_blank">BugAR-2936</a>  Mentions field has wrong position and covers the sent button when text field is expanded
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Sister PR of #1373

### Issue

When expanding the text field and you try to mention someone, suggestion list does not follow cursor position and it's displayed randomly

### Causes (Optional)

Refactor done in #1231 has invalidated some cursor position calculations.

### Solutions

Re-do it by adding new values and moving the dropdown menu from `Column` container to `Box`

#### How to Test

- Expand the text field by clicking on the top arrow
- Try to mention someone

### Attachments (Optional)

https://user-images.githubusercontent.com/18124919/214287513-d3f9a4aa-31ad-4995-933f-063449f0cac1.mp4 

https://user-images.githubusercontent.com/18124919/214287166-49df889b-ddd3-4a27-b303-f1e2714b98e1.mp4 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1377/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5561913)(5561913)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv0.0.1(5561913)(5561913)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt [hunk 1]
```diff
@@ -143,10 +146,12 @@ private const val MINIMUM_SCREEN_HEIGHT = 700
 private val DROP_DOWN_HEIGHT_SMALL = 150.dp
 private val DROP_DOWN_HEIGHT_LARGE = 200.dp
 private const val FIRST_LINE_INDEX = 0
-private const val HALF_SCREEN = 2.5
-private const val EIGHTY_PERCENT = 0.80
+private const val HALF_SCREEN = 0.45
+private const val SIXTY_PERCENT = 0.75
+private const val SEVENTY_PERCENT = 0.75
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt]: These two should be the same? Maybe we can reduce it and use only one of them?

**Review by saleniuk — APPROVED** [review]
Just one suggestion but overall well done for making it work again 👏

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/DropDownMentionsSuggestions.kt]: Oh! that's wrong. should not the same

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1377/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5619860)(5619860)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5619860)(5619860)-staging-compat.apk)

**Pull request merged**